### ISSUE TITLE
`r2` sends invalid HTTP requests if `Path` does not have a leading `/`.

### ISSUE THREAD

**Issue by dhermes** [blend/go-sdk#154]
To reproduce (at 480ad70d161406204ef58d3528b0ac17cf64bf0a), first listen to port 5007 with netcat and then run this sample program

```go
package main

import (
	"fmt"
	"time"

	"github.com/blend/go-sdk/r2"
)

func main() {
	r := r2.New(
		"",
		r2.OptScheme("http"),
		r2.OptHost("localhost:5007"),
		r2.OptPath("v1/resource"),
		r2.OptTimeout(5*time.Second),
	)
	res, err := r.Do()
	if err != nil {
		panic(err)
	}
	fmt.Printf("%#v\n", res)
}
```

After running, we have

```
$ nc -l 5007
GET v1/resource HTTP/1.1
Host: localhost:5007
User-Agent: Go-http-client/1.1
Accept-Encoding: gzip

```

I will dig deeper into this at a later time, but just wanted to get the repro posted here.

**Comment by dhermes** [comment]
The equivalent with `v2`

```go
package main

import (
	"fmt"
	"time"

	"github.com/blend/go-sdk/request"
)

func main() {
	r := request.New().
		WithScheme("http").
		WithHost("localhost:5007").
		WithPath("v1/resource").
		WithTimeout(5 * time.Second)
	res, err := r.Response()
	if err != nil {
		panic(err)
	}
	fmt.Printf("%#v\n", res)
}
```

produces

```
$ nc -l 5007
GET /v1/resource HTTP/1.1
Host: localhost:5007
User-Agent: Go-http-client/1.1
Accept-Encoding: gzip

```

**Comment by dhermes** [comment]
So I think the primary difference between v2 and v3 (which is #141 and is heavily related to #142 and #143) is the [usage of][1] `http.NewRequest()`

```go
func (r *Request) Request() (*http.Request, error) {
	...
	req, err := http.NewRequest(r.Method(), r.URL().String(), body)
	...
}
```

[1]: https://github.com/blend/go-sdk/blob/24dca5bdfc9ab3b9eb3ca17a3590b0ee2c7856cd/request/request.go#L675

**Comment by dhermes** [comment]
On particular [peculiarity][1] is that `net/url.URL` is supposed to paper over the missing slash issue. Both with and without a leading slash

```go
package main

import (
	"fmt"
	"net/url"
)

func main() {
	u := &url.URL{
		Scheme:   "https",
		Host:     "site.invalid",
		Path:     "v1/dogs",
		RawQuery: "a=b",
	}
	fmt.Printf("%q (%#v)\n", u.String(), u)
	u = &url.URL{
		Scheme:   "https",
		Host:     "site.invalid",
		Path:     "/v1/dogs",
		RawQuery: "a=b",
	}
	fmt.Printf("%q (%#v)\n", u.String(), u)
}
```

we get the same URL string even though `u.Path` differs

```
"https://site.invalid/v1/dogs?a=b" (&url.URL{... Path:"v1/dogs" ...})
"https://site.invalid/v1/dogs?a=b" (&url.URL{... Path:"/v1/dogs" ...})
```

[1]: https://play.golang.org/p/CjsRj_9KhBD

**Comment by dvdliao** [comment]
just happened for me too on existing api requests

**Comment by UjjwalAyyangar** [comment]
> So I think the primary difference between v2 and v3  is the [usage of](https://github.com/blend/go-sdk/blob/24dca5bdfc9ab3b9eb3ca17a3590b0ee2c7856cd/request/request.go#L675) `http.NewRequest()`
> 

That is right. This usage of `http.NewRequest` takes the `r.URL().String()` as a parameter, which handles the issue with leading `/`. 
(v1/resource -> .String() -> /v1/resource -> parseURL)

However in `r2`, the `.String()` function is absent from the execution flow between `OptPath` and `New` which led to this issue.

**Comment by UjjwalAyyangar** [comment]
> So I think the primary difference between v2 and v3  is the [usage of](https://github.com/blend/go-sdk/blob/24dca5bdfc9ab3b9eb3ca17a3590b0ee2c7856cd/request/request.go#L675) `http.NewRequest(r.Method(), r.URL().String(), body)`
> 

That is right. In this usage, the parameter  `r.URL().String()`  handles the issue with leading `/`. 
(v1/resource -> [.String()](https://golang.org/pkg/net/url/#URL.String) -> /v1/resource )

However in `r2`, the `.String()` function is absent from the execution flow between `OptPath` and `New` which led to this issue.
